snapshotter/app/service.go

```diff
@@ -134,17 +135,33 @@ func initSnapshotter(ctx context.Context, config config.Config) (snapshots.Snaps
 		if err != nil {
 			return nil, err
 		}
-		host, portstr, err := net.SplitHostPort(response.Address)
+		u, err := url.Parse(response.Address)
 		if err != nil {
 			return nil, err
 		}
-		port, err := strconv.ParseUint(portstr, base10, bits32)
+		host := u.Hostname()
```

Comment:
`net.SplitHostPort` will error if a port is not present in `response.Address` (which is the case per changes to `Resposne`), so opted to use net/url [`Parse`](https://pkg.go.dev/net/url#Parse) for validating address and [`Hostname()`](https://pkg.go.dev/net/url#URL.Hostname) for retrieving.